src/Data/FunctorF.purs

```diff
@@ -0,0 +1,17 @@
+module Data.FunctorF where
+
+import Data.NaturalTransformation (type (~>))
+
+-- | A type class for higher order functors; types of kind `(* -> *) -> * -> *`.
+-- |
+-- | The mapping operation `hoist` allows the inner functor to be changed. This
+-- | is similar to `map` for a `Functor`, only here we are transforming a
+-- | functor rather than a value, so correspondingly we perform this with a
+-- | natural transformation rather than a rank-1 function.
+-- |
+-- | Instances must satisfy the following laws:
+-- |
+-- | - Identity: `hoist id = id`
+-- | - Composition: `hoist (f <<< g) = hoist f <<< hoist g`
+class FunctorF t where
+  hoist :: forall f g. (f ~> g) -> t f ~> t g
```

Comment:
> Note the different constraints on the functions in the search results though. You can't apply those to f and g since they're quantified in hoist

I'm not really seeing what you mean. Isn't it just because haskell has implicit quantifiers. Only the `forall a.` is required for the natural transformation because it is rank-2, the `f` and `g` are not coming from the class.